tests/integration/targets/iam_user/tasks/main.yml

```diff
@@ -277,17 +357,25 @@
       that:
         - iam_user_update is not changed
 
-  # flakey, there is no waiter for login profiles
-  # Login Profile for User ansible-user-c cannot be modified while login profile is being created.
+  - name: update IAM password (check mode)
+    iam_user:
+      name: "{{ test_user3 }}"
+      password: "{{ test_new_password }}"
+      state: present
+    register: iam_user_update
+    check_mode: yes
+
+  - assert:
+      that:
+        - iam_user_update is changed
+
   - name: update IAM password
     iam_user:
       name: "{{ test_user3 }}"
       password: "{{ test_new_password }}"
       state: present
+      wait: False
     register: iam_user_update
-    until: iam_user_update.failed == false
-    delay: 3
-    retries: 5
```

Comment:
@jillr the point of the retry/wait was to resolve issues with login profile modifications not propagating in time. before adding any sort of wait, these tests would consistently fail when the until, delay, and retries parameters were removed from these tasks

Reply:
Ah ok, thank you!  In this case I think it's unlikely a real-world user would be doing this (immediately changing the profile for a brand new user).  What we're doing in the tests is pretty artificial, so I think it would be better to keep the retry here in the tests than add additional update calls to the code.  

It can be difficult to balance what we do in the test suite from what actually impacts our users.  In this case, the update could be changing a user's password.  Offhand I'm not aware of policies an account admin could set that limit the number of update_profile calls (ie; password changes) that can be run on a given user in a period of time, but I'd prefer to err on the side of being conservative.

Reply:
updated - thanks!